### PR TITLE
Helm values file support

### PR DESCRIPTION
**Pull request by metahertz**
allow using the `--var-file` option with Helm (as well as currently supported TF) frameworks, see documentation update in the commit for more detail.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by metahertz** [comment]
Do not merge, tests needed, comments welcome!

File: checkov/helm/runner.py [hunk 1]
```diff
@@ -114,8 +114,8 @@ def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=R
                 f"Processing chart found at: {chart_dir}, name: {chart_meta['name']}, version: {chart_meta['version']}")
             with tempfile.TemporaryDirectory() as target_dir:
                 # dependency list is nicer to parse than dependency update.
-                proc = subprocess.Popen([self.helm_command, 'dependency', 'list', chart_dir], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
-                o, e = proc.communicate()
+                proc1 = subprocess.Popen([self.helm_command, 'dependency', 'list', chart_dir], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
```
Comment (schosterbarak) [inline checkov/helm/runner.py:117]: @metahertz why changing from poc to poc1?

File: checkov/helm/runner.py [hunk 2]
```diff
@@ -126,16 +126,34 @@ def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=R
...
+                    try:
+                        # --dependency-update needed to pull in deps before templating.
+                        proc2 = subprocess.Popen([self.helm_command, 'template', '--dependency-update', chart_dir] + var_files_helm_formatted, stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
+                        o, e = proc2.communicate()
+                        logging.debug(
+                            f"Ran helm command to template chart output. Chart: {chart_meta['name']}. dir: {target_dir}. Output: {str(o, 'utf-8')}")
 
-                except Exception:
-                    logging.info(
+                    except Exception:
```
Comment (nimrodkor) [inline checkov/helm/runner.py:141]: @metahertz Missing an `as e:`?
```suggestion
                    except Exception as e:
```

File: checkov/helm/runner.py [hunk 3]
```diff
@@ -126,16 +126,34 @@ def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=R
...
+
+                else:
+                    try:
+                        # --dependency-update needed to pull in deps before templating.
+                        proc = subprocess.Popen([self.helm_command, 'template', '--dependency-update', chart_dir], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
+                        o, e = proc.communicate()
+                        logging.debug(
+                            f"Ran helm command to template chart output. Chart: {chart_meta['name']}. dir: {target_dir}. Output: {str(o, 'utf-8')}")
+                
+                    except Exception:
```
Comment (nimrodkor) [inline checkov/helm/runner.py:153]:
```suggestion
                    except Exception as e:
```

**Review by nimrodkor — COMMENTED** [review]
Very cool @metahertz !

File: checkov/helm/runner.py [hunk 4]
```diff
@@ -126,16 +126,34 @@ def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=R
 
                 self.parse_helm_dependency_output(o)
 
-                try:
-                    # --dependency-update needed to pull in deps before templating.
-                    proc = subprocess.Popen([self.helm_command, 'template', '--dependency-update', chart_dir], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
-                    o, e = proc.communicate()
-                    logging.debug(
-                        f"Ran helm command to template chart output. Chart: {chart_meta['name']}. dir: {target_dir}. Output: {str(o, 'utf-8')}")
+                if runner_filter.var_files != None:
```
Comment (gruebel) [inline checkov/helm/runner.py:129]:
```suggestion
                if runner_filter.var_files:
```
this way you also skip this part, when var_files is an empty list

File: checkov/helm/runner.py (hunk 2 above)
Comment (metahertz) [reply checkov/helm/runner.py:141]: Wasnt accessing the error, but 100% should have included! Thanks.

File: checkov/helm/runner.py (hunk 1 above)
Comment (metahertz) [reply checkov/helm/runner.py:117]: Good shout, not very readable! Thanks!

**Comment by metahertz** [comment]
Ready to Merge!

**Review by gruebel — APPROVED** [review]
🍾

**Pull request merged**